### PR TITLE
[VarCouldBeVal] fix overrides false positives

### PR DESCRIPTION
**Pull request by meriouma**
I'm not sure this is correct as this seems a simple check, but this fixes https://github.com/detekt/detekt/issues/4661 and does not break other tests.

### CODE REVIEW

**Review by marschwar — COMMENTED** [review]
Thank you for submitting the PR.

While I agree, that a `lateinit var` can never really become a `val`, I think the underlying issue here is the magic done by the framework. To detekt the variable never changes its value and therefor it could be val. Therefor changing your config to

```yaml
  VarCouldBeVal:
    active: true
    ignoreAnnotated: ['Inject']
```
would in my opinion also be a valid fix.

Long story short - I think your approach here is valid. The issue that a `private lateinit var` is never assigned a value should be handled by a different rule.

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/VarCouldBeValSpec.kt [hunk 1]
```diff
@@ -99,6 +99,15 @@ class VarCouldBeValSpec : Spek({
             """.trimIndent()
             assertThat(subject.compileAndLintWithContext(env, code)).hasSize(1)
         }
+
+        it("does not report lateinit variables") {
+            val code = """
+                class A {
+                    private lateinit var a = 1
```
[the remark this answers is not in the document]
Comment (meriouma) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/VarCouldBeValSpec.kt]: No, went to fast in a copy/paste. Should be fixed now.

**Comment by meriouma** [comment]
> Thank you for submitting the PR.
> 
> While I agree, that a `lateinit var` can never really become a `val`, I think the underlying issue here is the magic done by the framework. To detekt the variable never changes its value and therefor it could be val. Therefor changing your config to
> 
> ```yaml
>   VarCouldBeVal:
>     active: true
>     ignoreAnnotated: ['Inject']
> ```
> 
> would in my opinion also be a valid fix.
> 
> Long story short - I think your approach here is valid. The issue that a `private lateinit var` is never assigned a value should be handled by a different rule.

I ended up doing this workaround for the Inject annotation as this was the main issue in one of our project. But there are many frameworks that rely on an annotation and a `lateinit var` field, like Mockito. So instead of ignoring all the annotations, I thought `lateinit` could be ignored by the rule. I believe there's already the rule `LateinitUsage` to check for lateinit anyway?

**Comment by BraisGabin** [comment]
I don't think that we should ignore `lateinit var` by default. They are `var` so if they aren't assigned they should be flagged. We have methods to disable properties with annotations. At most I would say to add a configuration, but i would like to see a user case where it is needed. Dagger or mockito doesn't need this to work.

**Comment by marschwar** [comment]
@meriouma Could you please change this PR to address the issue with `override` only. I believe that part is not controversial. 

I can see the arguments on either side. Since there is an easy workaround for your specific problem I would also not change it at the moment. Maybe this becomes a pattern and we could change it in the future.

**Comment by meriouma** [comment]
@marschwar Done!

**Comment by marschwar** [comment]
@meriouma Thanks again for picking this up. I would like us to merge #4575 first to avoid a moving target there and would ask your to adopt your test case to junit if that is ok. I am sorry for the inconvenience.

**Comment by meriouma** [comment]
@marschwar Sure, no problem. I can rebase and rewrite the test case once #4575 is merged.

**Review by schalkms — COMMENTED** [review]
PR https://github.com/detekt/detekt/pull/4575 has been merged.

**Comment by meriouma** [comment]
I've rebase and updated the test.

**Review by BraisGabin — APPROVED** [review]
Thanks!

**Pull request merged**